python/cuml/dask/common/utils.py

```diff
@@ -153,12 +160,106 @@ def patch_cupy_sparse_serialization(client):
 
     client : dask.distributed.Client client to use
     """
+
+    from distributed.protocol import register_generic
+
     def patch_func():
         def serialize_mat_descriptor(m):
             return cp.cupy.cusparse.MatDescriptor.create, ()
 
+        register_generic(Base)
+
         copyreg.pickle(cp.cupy.cusparse.MatDescriptor,
                        serialize_mat_descriptor)
 
     patch_func()
     client.run(patch_func)
+
+
+class MultiHolderLock:
+    """
+    A per-process synchronization lock allowing multiple concurrent holders
+    at any one time. This is used in situations where resources might be
+    limited and it's important that the number of concurrent users of
+    the resources are constained.
+
+    This lock is serializable, but relies on a Python threading.Lock
+    underneath to properly synchronize internal state across threads.
+    Note that this lock is only intended to be used per-process and
+    the underlying threading.Lock will not be serialized.
+    """
+
+    def __init__(self, n):
+        """
+        Initialize the lock
+        :param n : integer the maximum number of concurrent holders
```

Comment:
wrong comment style?

Reply:
Yep! My intellij does this by default and sometimes I don't catch it before I update the pydocs.